robot/impl/resource_manager.go

```diff
@@ -1023,14 +1030,59 @@ func (manager *resourceManager) markResourcesRemoved(
 func (manager *resourceManager) removeOrphanedResources(ctx context.Context,
 	rNames []resource.Name,
 ) {
-	// Ignore returned resources to close, as removeMarkedAndClose will handle.
 	manager.markResourcesRemoved(rNames, nil)
-	if _, err := manager.removeMarkedAndClose(ctx, nil); err != nil {
+	if err := manager.removeMarkedAndClose(ctx, nil); err != nil {
 		manager.logger.Errorw("error removing and closing marked resources",
 			"error", err)
 	}
 }
 
+// createConfig will create a config.Config based on the current state of the
```

Comment:
should there be a comment about how this is not necessarily reflecting the user config to the robot due to builtin services?